src/profiles.jl

```diff
@@ -512,36 +512,154 @@ function add_metadata!(commp::CommunityProfile, samplecol::Symbol, md; overwrite
     return nothing
 end
 
+"""
+    set!(commp::CommunityProfile, sample::AbstractString, prop::Symbol, val)
+
+Update or insert a value `val` to the `sample` in the CommunityProfile `commp` using a Symbol `prop`. 
+
+Examples
+≡≡≡≡≡≡≡≡≡≡
+
+```jldoctest
+julia> set!(comm, "sample1", :something, 1.0)
+
+julia> first(metadata(comm))[:something]
+1.0
+ ```
+"""
 function set!(commp::CommunityProfile, sample::AbstractString, prop::Symbol, val)
     sample = samples(commp, sample)
     prop in _restricted_fields(sample) && error("Cannot set! $prop for $(typeof(sample)).")
     set!(sample.metadata, prop, val)
     return sample
 end
 
+"""
+    unset!(commp::CommunityProfile, sample::AbstractString, prop::Symbol)
+
+Delete a `sample` from CommunityProfile `commp` using the Symbol `prop` if it exists, or throw an error otherwise.
+
+Examples
+≡≡≡≡≡≡≡≡≡≡
+
+```jldoctest
+julia> unset!(comm, "sample1", :something) 
+
+julia> !haskey(comm, "sample1", :something)
+true
+ ```
+"""
 function unset!(commp::CommunityProfile, sample::AbstractString, prop::Symbol)
     sample = samples(commp, sample)
     prop in _restricted_fields(sample) && error("Cannot unset! $prop for $(typeof(sample)).")
     unset!(sample.metadata, prop)
     return sample
 end
 
+"""
+    insert!(commp::CommunityProfile, sample::AbstractString, prop::Symbol, val)
+
+Insert a value `val` to the `sample` in a CommunityProfile `commp` using a Symbol `prop`.
+
+Examples
+≡≡≡≡≡≡≡≡≡≡
+
+```jldoctest
+julia> insert!(comm, "sample1", :something, 3.0) 
+
+julia> get(comm, "sample1", :something, 3.0)
+3.0
+ ```
+"""
 function insert!(commp::CommunityProfile, sample::AbstractString, prop::Symbol, val)
     sample = samples(commp, sample)
     prop in _restricted_fields(sample) && error("Cannot insert! $prop for $(typeof(sample)).")
     insert!(sample.metadata, prop, val)
     return sample
 end
 
+"""
+    delete!(commp::CommunityProfile, sample::AbstractString, prop::Symbol)
+
+Delete a `sample` from CommunityProfile `commp` using the Symbol `prop` if it exists, or throw an error otherwise.
+
+Examples
+≡≡≡≡≡≡≡≡≡≡
+
+```jldoctest
+julia> delete!(comm, "sample1", :something) 
+
+julia> !haskey(comm, "sample1", :something)
+true
+ ```
+"""
 function delete!(commp::CommunityProfile, sample::AbstractString, prop::Symbol)
     sample = samples(commp, sample)
     prop in _restricted_fields(sample) && error("Cannot delete! $prop for $(typeof(sample)).")
     delete!(sample.metadata, prop)
     return sample
 end
 
+"""
+    keys(commp::CommunityProfile, sample::AbstractString)
+
+Return an iterator over all keys of samples in a CommunityProfile. collect(keys(a)) returns an array of keys. When the keys are stored internally in a hash table, as is
```

Comment:
Please use [semantic line breaks](http://sembr.org) (SEMBR) - eg at least newlines at the end of sentences